hnn_core/opt_toy_example/general.py

```diff
@@ -0,0 +1,365 @@
+                       predicted_params,
+                       _update_params,
+                       obj_values,
+                       scale_factor,
+                       smooth_window_len,
+                       tstop,
+                       target)
+
+    opt_results = gp_minimize(func=_obj_func,
+                              dimensions=constraints,
+                              acq_func='EI',
+                              n_calls=max_iter,
+                              x0=list(initial_params.values()))
+
+    # get optimized params
+    opt_params = opt_results.x
+
+    # get objective values
+    obj = [np.min(obj_values[:i]) for i in range(1, max_iter + 1)]
+
+    # get optimized net
+    param_dict = _update_params(initial_params, opt_params)
+    net_ = net.copy()
+    set_params(net_, param_dict)
+
+    return opt_params, obj, net_
+
+
+def _run_opt_cobyla(net, constraints, initial_params, set_params, obj_fun,
+                    scale_factor, smooth_window_len, tstop, max_iter,
+                    target):
+    """
+    Runs optimization routine with fmin_cobyla optimizer.
+
+    Parameters
+    ----------
+    net : Network
+        The network object.
+    constraints : dict
+        Parameter constraints in solver-specific format.
+    initial_params : dict
+        Keys are parameter names, values are initial parameters..
+    set_params : func
+        User-defined function that sets parameters in network drives.
+    obj_fun : func
+        The objective function.
+    scale_factor : float
+        The dipole scale factor.
+    smooth_window_len : float
+        The smooth window length.
+    tstop : float
+        The simulated dipole's duration.
+    max_iter : int
+        Number of calls the optimizer makes.
+    target : ndarray, None
+        The recorded dipole. The default is None.
+
+    Returns
+    -------
+    opt_params : list
+        Optimized parameters.
+    obj : list
+        Objective values.
+    net_ : Network
+        Optimized network object.
+    """
+
+    obj_values = list()
+
+    def _obj_func(predicted_params):
+        return obj_fun(net,
+                       initial_params,
+                       set_params,
+                       predicted_params,
+                       _update_params,
+                       obj_values,
+                       scale_factor,
+                       smooth_window_len,
+                       tstop,
+                       target)
+
+    opt_results = fmin_cobyla(_obj_func,
+                              cons=constraints,
+                              rhobeg=0.1,
+                              rhoend=1e-4,
+                              x0=list(initial_params.values()),
+                              maxfun=max_iter,
+                              catol=0.0)
+
+    # get optimized params
+    opt_params = opt_results
+
+    # get objective values
+    obj = [np.min(obj_values[:i]) for i in range(1, max_iter + 1)]
```

Comment:
```suggestion
    obj = [np.min(obj_values[:idx]) for idx in range(1, max_iter + 1)]
```

avoid single letter variable names when you can ... they are hard to find and replace + not very informative